src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -434,13 +455,21 @@ private async Task UpdateCatagory(Guid categoryId, ExceptionCategorySettings cat
             {
                 ExceptionBreakpointStates newCategoryState = updates.NewCategoryState.Value;
                 categorySettings.CategoryState = newCategoryState;
+
+                if (categorySettings.CurrentRules.TryGetValue("*", out _) ||
+                    categorySettings.CurrentRules.TryGetValue("winrt::hresult_error", out _))
+                {
+                    await _commandFactory.RemoveExceptionBreakpoint(categoryId, categorySettings.CurrentRules.Values);
+                }
+
                 categorySettings.CurrentRules.Clear();
 
-                IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, null, newCategoryState);
-                if (newCategoryState != ExceptionBreakpointStates.None)
+                // IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, null, newCategoryState);
+                if (newCategoryState == ExceptionBreakpointStates.BreakThrown) // only set catchpoint on BreakThrown
                 {
-                    ulong breakpointId = breakpointIds.Single();
-                    categorySettings.CurrentRules.Add("*", breakpointId);
+                    IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, null, newCategoryState);
+                    ulong singleBreakpointId = breakpointIds.Single();
```

Comment:
Here is where you have the breakpoint id (long), category guid, name, and state.